tests/test_cli_compile.py

```diff
@@ -1063,6 +1145,25 @@ def test_annotate_option(pip_conf, runner, option, expected):
     assert out.exit_code == 0
 
 
+def test_annotation_relative_paths(pip_conf, runner):
```

Comment:
Would be nice to have a separate test for `pip-compile setup.py` and check `(path/to/setup.py)` annotations.

Reply:
I'd like to do something like the following, but during the test it fails because it can't install `setuptools`.

```
ERROR: ERROR: Could not find a version that satisfies the requirement setuptools
ERROR: ERROR: No matching distribution found for setuptools
```

It's not clear to me right now what the proper resolution is, any advice?

```python
@pytest.mark.parametrize(
    ("input_filename", "input_content"),
    (
        ("requirements.in", "small-fake-a==0.1"),
        ("setup.py", (
            "from setuptools import setup\n"
            "setup(name='fake-setuptools-a', install_requires=['small-fake-a==0.1'])"
        )),
    ),
)
def test_annotation_relative_paths(pip_conf, runner, input_filename, input_content):
    """
    Annotations referencing reqs.in or setup.py files use paths relative to the reqs.txt.
    """
    in_path = os.path.abspath(input_filename)
    txt_path = os.path.abspath(os.path.join("deeper", "requirements.txt"))
    rel_path = os.path.relpath(in_path, os.path.dirname(txt_path))

    with open(in_path, "w") as req_in:
        req_in.write(input_content)

    os.mkdir("deeper")
    out = runner.invoke(cli, ["-o", txt_path])

    assert out.exit_code == 0
    with open(txt_path) as req_txt:
        assert " # via -r " + rel_path in req_txt.read()  # may need another param here
```

Reply:
@atugushev @webknjaz 

Can you please help me with the test in this comment thread? It's not clear to me why some tests don't have trouble installing setuptools while this one does.

Reply:
Also pinging @vphilippon to request a look at this test, as I'm eager to get this test working.

Reply:
I'm not entirely sure but my first bet would be to check the env vars and the fixture hierarchy used in the working test vs the broken one. Also, try to understand if there's any "autouse" fixtures involved.

Reply:
By the way, I strongly recommend using the built-in `tmp_path` fixture for test layouts on disk because it gets autocleanedup upon test completion + makes sure you don't influence the global state by touching something in the source dir.

Reply:
@AndydeCleyre FWIW it's best if you commit the changes to this PR so we could see *how* it breaks in more detail without having to check out your branch to test locally which is more time-consuming (and therefore is traditionally postponed more often).

Reply:
@webknjaz Thanks very much, the broken changes are up, and `tmp_path` does improve the test.

Reply:
Thanks! It didn't make the reason obvious, unfortunately. I'll have to get your branch locally and inject some breakpoints maybe. One of the things to do is to add `breakpoint()` at the beginning of the test and run `pytest` with `-s` so you'll be able to inspect locals and env vars in runtime.